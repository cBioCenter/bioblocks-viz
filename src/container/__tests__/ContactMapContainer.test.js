"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var enzyme_1 = require("enzyme");
var React = require("react");
var container_1 = require("~bioblocks-viz~/container");
var data_1 = require("~bioblocks-viz~/data");
var test_1 = require("~bioblocks-viz~/test");
describe('ContactMapContainer', function () {
    var emptyData;
    var sampleCorrectPredictedContacts;
    var sampleData;
    var sampleIncorrectPredictedContacts;
    var sampleObservedContacts;
    var sampleOutOfLinearDistContacts;
    var uniqueScores;
    beforeEach(function () {
        emptyData = {
            couplingScores: new data_1.CouplingContainer(),
            secondaryStructures: [],
        };
        // Translated from example1/coupling_scores.csv
        sampleCorrectPredictedContacts = [generateCouplingScore(56, 50, 2.4)];
        sampleIncorrectPredictedContacts = [generateCouplingScore(42, 50, 20.4)];
        sampleObservedContacts = tslib_1.__spread(sampleCorrectPredictedContacts, [generateCouplingScore(41, 52, 1.3)]);
        sampleOutOfLinearDistContacts = [
            generateCouplingScore(45, 46, 1.3),
            generateCouplingScore(44, 45, 1.3),
            generateCouplingScore(56, 57, 1.3),
        ];
        uniqueScores = new Set(Array.from(tslib_1.__spread(sampleCorrectPredictedContacts, sampleIncorrectPredictedContacts, sampleObservedContacts, sampleOutOfLinearDistContacts)));
        sampleData = {
            couplingScores: new data_1.CouplingContainer(Array.from(uniqueScores)),
            secondaryStructures: [[new data_1.Bioblocks1DSection('C', 30, 31)]],
        };
    });
    var generateCouplingScore = function (i, j, dist, extra) { return (tslib_1.__assign({ dist: dist,
        i: i,
        j: j }, extra)); };
    describe('Snapshots', function () {
        it('Should match existing snapshot when given no data.', function () {
            expect(enzyme_1.shallow(React.createElement(container_1.ContactMapContainerClass, null))).toMatchSnapshot();
        });
        it('Should match existing snapshot when given empty data.', function () {
            expect(enzyme_1.shallow(React.createElement(container_1.ContactMapContainerClass, { data: emptyData }))).toMatchSnapshot();
        });
        it('Should match snapshot when locked residues are added.', function () { return tslib_1.__awaiter(void 0, void 0, void 0, function () {
            var wrapper, expectedSelectedPoints;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, test_1.getAsyncShallowComponent(React.createElement(container_1.ContactMapContainerClass, null))];
                    case 1:
                        wrapper = _a.sent();
                        expectedSelectedPoints = {
                            '37,46': [37, 46],
                            8: [8],
                        };
                        wrapper.setProps({
                            lockedResiduePairs: expectedSelectedPoints,
                        });
                        wrapper.update();
                        expect(wrapper).toMatchSnapshot();
                        return [2 /*return*/];
                }
            });
        }); });
    });
    describe('Sliders', function () {
        it('Should update linear distance filter when appropriate slider is updated.', function () {
            var wrapper = enzyme_1.shallow(React.createElement(container_1.ContactMapContainerClass, { data: sampleData }));
            var instance = wrapper.instance();
            var expected = 10;
            expect(instance.state.linearDistFilter).not.toBe(expected);
            instance.onLinearDistFilterChange()(expected);
            instance.forceUpdate();
            expect(instance.state.linearDistFilter).toBe(expected);
        });
        it('Should update number of predicted contacts to show when appropriate slider is updated.', function () {
            var wrapper = enzyme_1.shallow(React.createElement(container_1.ContactMapContainerClass, { data: sampleData }));
            var instance = wrapper.instance();
            var expectedCount = 50;
            expect(instance.state.numPredictionsToShow).not.toBe(expectedCount);
            instance.onNumPredictionsToShowChange()(expectedCount);
            wrapper.update();
            expect(instance.state.numPredictionsToShow).toBe(expectedCount);
        });
        it('Should update # of predicted contacts to show when appropriate slider is updated.', function () {
            var wrapper = enzyme_1.shallow(React.createElement(container_1.ContactMapContainerClass, { data: sampleData }));
            var instance = wrapper.instance();
            var expected = 20;
            expect(instance.state.numPredictionsToShow).not.toBe(expected);
            instance.onNumPredictionsToShowChange()(expected);
            wrapper.update();
            expect(instance.state.numPredictionsToShow).toBe(expected);
        });
    });
    it('Should update number of predictions to show when new value is received.', function () {
        var expected = 28;
        var wrapper = enzyme_1.shallow(React.createElement(container_1.ContactMapContainerClass, { data: emptyData }));
        expect(wrapper.state('numPredictionsToShow')).not.toBe(expected);
        wrapper.setState({
            numPredictionsToShow: expected,
        });
        expect(wrapper.state('numPredictionsToShow')).toBe(expected);
    });
    it('Should update points to plot when new data is provided.', function () {
        var wrapper = enzyme_1.shallow(React.createElement(container_1.ContactMapContainerClass, { data: emptyData }));
        wrapper.setProps({
            data: sampleData,
        });
        var state = wrapper.instance().state;
        expect(state.pointsToPlot).not.toEqual([]);
        expect(state.pointsToPlot).toMatchSnapshot();
    });
});
//# sourceMappingURL=ContactMapContainer.test.js.map