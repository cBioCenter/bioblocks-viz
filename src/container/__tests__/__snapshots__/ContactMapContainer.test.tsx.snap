// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactMapContainer Should update points to plot when new data is provided. 1`] = `
Array [
  Object {
    "hoverinfo": "text",
    "marker": Object {
      "color": Array [
        "#000000",
        "#000000",
        "#000000",
        "#000000",
        "#000000",
        "#000000",
      ],
    },
    "mode": "lines+markers",
    "name": "Inferred Contact",
    "nodeSize": 4,
    "points": Array [
      Object {
        "dist": 1.3,
        "i": 41,
        "j": 52,
      },
      Object {
        "dist": 20.4,
        "i": 42,
        "j": 50,
      },
      Object {
        "dist": 2.4,
        "i": 50,
        "j": 56,
      },
    ],
    "subtitle": "(N=28, L=57)",
    "text": Array [
      "(41, 52)",
      "(42, 50)",
      "(50, 56)",
    ],
  },
  Object {
    "hoverinfo": "text",
    "marker": Object {
      "color": Array [
        "#ff0000",
        "#ff0000",
        "#ff0000",
        "#ff0000",
      ],
    },
    "mode": "lines+markers",
    "name": "Inferred Contact Agrees with PDB Contact",
    "nodeSize": 6,
    "points": Array [
      Object {
        "dist": 1.3,
        "i": 41,
        "j": 52,
      },
      Object {
        "dist": 2.4,
        "i": 50,
        "j": 56,
      },
    ],
    "subtitle": "(N=2, 66.7%)",
    "text": Array [
      "(41, 52)",
      "(50, 56)",
    ],
  },
]
`;

exports[`ContactMapContainer Snapshots Should match existing snapshot when given empty data. 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <div
    className="ContactMapContainer"
  >
    <ContactMapComponent
      addHoveredResidues={[Function]}
      addHoveredSecondaryStructure={[Function]}
      addSelectedSecondaryStructure={[Function]}
      agreementColor="#ff0000"
      allColor="#000000"
      configurations={
        Array [
          Object {
            "marks": Object {
              "0": "0",
              "1": "L",
              "2": "2L",
              "3": "3L",
            },
            "name": "# Couplings to Display",
            "onChange": [Function],
            "type": "SLIDER",
            "values": Object {
              "current": 0,
              "defaultValue": 0,
              "max": 3,
              "min": 0,
            },
          },
          Object {
            "name": "Linear Distance Filter (|i - j|)",
            "onChange": [Function],
            "type": "SLIDER",
            "values": Object {
              "current": 5,
              "defaultValue": 5,
              "max": 10,
              "min": 1,
            },
          },
          Object {
            "name": "Minimum Probability",
            "onChange": [Function],
            "step": 0.01,
            "type": "SLIDER",
            "values": Object {
              "current": 0.9,
              "defaultValue": 0.9,
              "max": 1,
              "min": 0,
            },
          },
        ]
      }
      data={
        Object {
          "couplingScores": CouplingContainer {
            "colCounter": 0,
            "contacts": Array [],
            "derivedFromCouplingScoresFlag": true,
            "getCouplingScore": [Function],
            "includes": [Function],
            "indexRange": Object {
              "max": 1,
              "min": 1,
            },
            "rowCounter": 0,
            "scoreSource": "cn",
            "totalStoredContacts": 0,
          },
          "pdbData": undefined,
          "secondaryStructures": Array [],
        }
      }
      enableSliders={true}
      filters={Array []}
      formattedPoints={
        Array [
          Object {
            "hoverinfo": "text",
            "marker": Object {
              "color": Array [],
            },
            "mode": "lines+markers",
            "name": "Inferred Contact",
            "nodeSize": 4,
            "points": Array [],
            "subtitle": "(N=0, L=1)",
            "text": Array [],
          },
          Object {
            "hoverinfo": "text",
            "marker": Object {
              "color": Array [],
            },
            "mode": "lines+markers",
            "name": "Inferred Contact Agrees with PDB Contact",
            "nodeSize": 6,
            "points": Array [],
            "subtitle": "(N=0, NaN%)",
            "text": Array [],
          },
        ]
      }
      height="100%"
      highlightColor="#ff8800"
      hoveredResidues={Array []}
      hoveredSecondaryStructures={Array []}
      isDataLoading={false}
      lockedResiduePairs={Object {}}
      observedColor="#0000ff"
      onBoxSelection={[Function]}
      removeAllLockedResiduePairs={[Function]}
      removeAllSelectedSecondaryStructures={[Function]}
      removeHoveredResidues={[Function]}
      removeHoveredSecondaryStructure={[Function]}
      removeSecondaryStructure={[Function]}
      selectedSecondaryStructures={Array []}
      showConfigurations={true}
      toggleLockedResiduePair={[Function]}
      width="100%"
    />
  </div>
</Provider>
`;

exports[`ContactMapContainer Snapshots Should match existing snapshot when given no data. 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <div
    className="ContactMapContainer"
  >
    <ContactMapComponent
      addHoveredResidues={[Function]}
      addHoveredSecondaryStructure={[Function]}
      addSelectedSecondaryStructure={[Function]}
      agreementColor="#ff0000"
      allColor="#000000"
      configurations={
        Array [
          Object {
            "marks": Object {
              "0": "0",
              "1": "L",
              "2": "2L",
              "3": "3L",
            },
            "name": "# Couplings to Display",
            "onChange": [Function],
            "type": "SLIDER",
            "values": Object {
              "current": 0,
              "defaultValue": 0,
              "max": 3,
              "min": 0,
            },
          },
          Object {
            "name": "Linear Distance Filter (|i - j|)",
            "onChange": [Function],
            "type": "SLIDER",
            "values": Object {
              "current": 5,
              "defaultValue": 5,
              "max": 10,
              "min": 1,
            },
          },
          Object {
            "name": "Minimum Probability",
            "onChange": [Function],
            "step": 0.01,
            "type": "SLIDER",
            "values": Object {
              "current": 0.9,
              "defaultValue": 0.9,
              "max": 1,
              "min": 0,
            },
          },
        ]
      }
      data={
        Object {
          "couplingScores": CouplingContainer {
            "colCounter": 0,
            "contacts": Array [],
            "derivedFromCouplingScoresFlag": true,
            "getCouplingScore": [Function],
            "includes": [Function],
            "indexRange": Object {
              "max": 1,
              "min": 1,
            },
            "rowCounter": 0,
            "scoreSource": "cn",
            "totalStoredContacts": 0,
          },
          "pdbData": undefined,
          "secondaryStructures": Array [],
        }
      }
      enableSliders={true}
      filters={Array []}
      formattedPoints={
        Array [
          Object {
            "hoverinfo": "text",
            "marker": Object {
              "color": Array [],
            },
            "mode": "lines+markers",
            "name": "Inferred Contact",
            "nodeSize": 4,
            "points": Array [],
            "subtitle": "(N=0, L=1)",
            "text": Array [],
          },
          Object {
            "hoverinfo": "text",
            "marker": Object {
              "color": Array [],
            },
            "mode": "lines+markers",
            "name": "Inferred Contact Agrees with PDB Contact",
            "nodeSize": 6,
            "points": Array [],
            "subtitle": "(N=0, NaN%)",
            "text": Array [],
          },
        ]
      }
      height="100%"
      highlightColor="#ff8800"
      hoveredResidues={Array []}
      hoveredSecondaryStructures={Array []}
      isDataLoading={false}
      lockedResiduePairs={Object {}}
      observedColor="#0000ff"
      onBoxSelection={[Function]}
      removeAllLockedResiduePairs={[Function]}
      removeAllSelectedSecondaryStructures={[Function]}
      removeHoveredResidues={[Function]}
      removeHoveredSecondaryStructure={[Function]}
      removeSecondaryStructure={[Function]}
      selectedSecondaryStructures={Array []}
      showConfigurations={true}
      toggleLockedResiduePair={[Function]}
      width="100%"
    />
  </div>
</Provider>
`;

exports[`ContactMapContainer Snapshots Should match snapshot when locked residues are added. 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <div
    className="ContactMapContainer"
  >
    <ContactMapComponent
      addHoveredResidues={[Function]}
      addHoveredSecondaryStructure={[Function]}
      addSelectedSecondaryStructure={[Function]}
      agreementColor="#ff0000"
      allColor="#000000"
      configurations={
        Array [
          Object {
            "marks": Object {
              "0": "0",
              "1": "L",
              "2": "2L",
              "3": "3L",
            },
            "name": "# Couplings to Display",
            "onChange": [Function],
            "type": "SLIDER",
            "values": Object {
              "current": 0,
              "defaultValue": 0,
              "max": 3,
              "min": 0,
            },
          },
          Object {
            "name": "Linear Distance Filter (|i - j|)",
            "onChange": [Function],
            "type": "SLIDER",
            "values": Object {
              "current": 5,
              "defaultValue": 5,
              "max": 10,
              "min": 1,
            },
          },
          Object {
            "name": "Minimum Probability",
            "onChange": [Function],
            "step": 0.01,
            "type": "SLIDER",
            "values": Object {
              "current": 0.9,
              "defaultValue": 0.9,
              "max": 1,
              "min": 0,
            },
          },
        ]
      }
      data={
        Object {
          "couplingScores": CouplingContainer {
            "colCounter": 0,
            "contacts": Array [],
            "derivedFromCouplingScoresFlag": true,
            "getCouplingScore": [Function],
            "includes": [Function],
            "indexRange": Object {
              "max": 1,
              "min": 1,
            },
            "rowCounter": 0,
            "scoreSource": "cn",
            "totalStoredContacts": 0,
          },
          "pdbData": undefined,
          "secondaryStructures": Array [],
        }
      }
      enableSliders={true}
      filters={Array []}
      formattedPoints={
        Array [
          Object {
            "hoverinfo": "text",
            "marker": Object {
              "color": Array [],
            },
            "mode": "lines+markers",
            "name": "Inferred Contact",
            "nodeSize": 4,
            "points": Array [],
            "subtitle": "(N=0, L=1)",
            "text": Array [],
          },
          Object {
            "hoverinfo": "text",
            "marker": Object {
              "color": Array [],
            },
            "mode": "lines+markers",
            "name": "Inferred Contact Agrees with PDB Contact",
            "nodeSize": 6,
            "points": Array [],
            "subtitle": "(N=0, NaN%)",
            "text": Array [],
          },
        ]
      }
      height="100%"
      highlightColor="#ff8800"
      hoveredResidues={Array []}
      hoveredSecondaryStructures={Array []}
      isDataLoading={false}
      lockedResiduePairs={
        Object {
          "37,46": Array [
            37,
            46,
          ],
          "8": Array [
            8,
          ],
        }
      }
      observedColor="#0000ff"
      onBoxSelection={[Function]}
      removeAllLockedResiduePairs={[Function]}
      removeAllSelectedSecondaryStructures={[Function]}
      removeHoveredResidues={[Function]}
      removeHoveredSecondaryStructure={[Function]}
      removeSecondaryStructure={[Function]}
      selectedSecondaryStructures={Array []}
      showConfigurations={true}
      toggleLockedResiduePair={[Function]}
      width="100%"
    />
  </div>
</Provider>
`;
